empiric-package/empiric/publisher/client.py

```diff
@@ -33,36 +42,35 @@ def __init__(
     async def _fetch_contracts(self):
         await self._fetch_base_contracts()
 
-    async def update_publisher_address(self, new_address, publisher=None):
-        if self.publisher is None and publisher is None:
+    async def update_publisher_address(self, new_address, publisher=None) -> HEX_STR:
+        publisher = publisher or self.publisher
+        if publisher is None:
             raise ValueError(
                 "No publisher provided at method call or instantiation, but need publisher ID to update address"
             )
-        elif publisher is None:
-            publisher = self.publisher
 
         result = await self.send_transaction(
             self.publisher_registry_address,
             "update_publisher_address",
             [publisher, new_address],
         )
-        print(f"Updated publisher address with transaction {result}")
+        LOGGER.info(f"Updated publisher address with transaction {result}")
 
         return result
 
-    async def publish(self, entry):
+    async def publish(self, entry) -> HEX_STR:
         result = await self.send_transaction(
             self.oracle_controller_address,
             "publish_entry",
             serialize_entry(entry),
         )
-        print(f"Updated entry with transaction {result}")
+        LOGGER.info(f"Updated entry with transaction {result}")
```

Comment:
Beautiful, tackled the next issue I've been itching to get to in the backlog :) https://www.notion.so/42labs-xyz/Move-print-to-logging-in-Python-c2062ced483d476a80fc103ac5d38a7f